### PR TITLE
set common protobuf files to compile with other tags in debug

### LINKED ISSUE #3431: debug build fails on fresh ubuntu 18.04 install
<b>System Information:</b>
Ubuntu 18.04 x64
Qt 5.9.5
gcc 7.3.0
cmake 3.10.2
protobuf-compiler 3.0.0-9.1
building from master
__________________________________________________________________________________________

Building fails on the generated protobuf source files at a warning (due to the -Werror flag) when using the debug build with `-DCMAKE_BUILD_TYPE=Debug`.
We can't fix this inside the source files because they are generated by the protobuf compiler instead.
Setting -Wno-error=unused-parameter -Wno-error=misleading-indentation fixes this, but these errors are preferred to trigger on normal source files.

The first generated error is as follows, there is about a hundred similar ones:
```
/Cockatrice/build/common/pb/admin_commands.pb.cc: In member function 'virtual google::protobuf::uint8* AdminCommand::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const':
/Cockatrice/build/common/pb/admin_commands.pb.cc:367:10: error: unused parameter 'deterministic' [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
          ^~~~~~~~~~~~~
```

### PR DESCRIPTION
**Pull request by ebbit1q**
Force the same flags as release on debug builds for the generated protobuf sources in common.
fixes #3431

### CODE REVIEW

**Comment by ebbit1q** [comment]
I thought that we trusted protobuf to make good code so we didn't need to check it, I implemented your changes

**Comment by ZeldaZach** [comment]
Deferring review to @ctrlaltca since he deals with protobuf more often than I do

**Comment by ctrlaltca** [comment]
> I thought that we trusted protobuf to make good code so we didn't need to check it, I implemented your changes

Thank you for addressing my feedback.
The Protobuf guys are quite fast at fixing their code, unfortunately the ones to blame here are the ubuntu (debian?) maintainers that never cared about applying the patch on their package.

**Comment by ebbit1q** [comment]
What I mean is the code is automatically generated, what is the point of checking for warnings if you can't resolve the issues?

File: common/pb/CMakeLists.txt [hunk 1]
```diff
@@ -168,3 +168,16 @@ if (UNIX)
     set(cockatrice_protocol_LIBS ${cockatrice_protocol_LIBS} -lpthread)
 endif (UNIX)
 target_link_libraries(cockatrice_protocol ${cockatrice_protocol_LIBS})
+
+# ubuntu uses an outdated package for protobuf1
+exec_program(
+    protoc
```
[the remark this answers is not in the document]
Comment (ebbit1q) [reply common/pb/CMakeLists.txt]: Protobuf_VERSION is apparently available instead, are they case insensitive? cmake docs aren't exactly clear.

**Comment by ctrlaltca** [comment]
> if(${Protobuf_VERSION} VERSION_LESS "3.5.0")

Please don't check for <= 3.5.0. There are several other distros out there shipping versions older than 3.5.0 (2.6 or 3.x) that doesn't need this. Only the debian/ubuntu 3.0.0-x package is affected by this issue.

> Simply appending -Wno-error would also work

This is not optimal, since it would also mute any other future warning caused by bugs in our pb's code.

File: common/pb/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (ctrlaltca) [reply common/pb/CMakeLists.txt]: Yes, variable names are case sensitive in cmake: https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#variables

**Comment by ebbit1q** [comment]
"VERSION_LESS 3.5.0" means "< 3.5.0" not "<= 3.5.0".
I did my best to check this, the specific warnings generated were fixed only in version 3.5 and no earlier, so anyone building with a protobuf version lower than this on gcc7 with "-Wall -Wextra -Werror" would have it fail.

**Comment by ctrlaltca** [comment]
Here's the commit that fixes the `deterministic` bug: https://github.com/protocolbuffers/protobuf/commit/08b1c718e437041bfe0e6a28611621896e4fe904, it has been merged in protobuf v3.1.0-alpha-1.
The bug is only present in 3.0.x versions, since the fix commit has been merged but 3.0.3 was never released: https://github.com/protocolbuffers/protobuf/compare/v3.0.2...3.0.x

**Comment by ebbit1q** [comment]
Cool, I was searching for that commit but only found something in the the release notes.

**Comment by ebbit1q** [comment]
all versions of ubuntu are

**Comment by tooomm** [comment]
Or link to the bug so they get all information and the status there?
https://bugs.launchpad.net/ubuntu/+source/protobuf/+bug/1643959

**Comment by ctrlaltca** [comment]
Merged, thank you!

**Pull request merged**